components/main_page/commons/main_page_layout.lua

```diff
@@ -65,44 +65,50 @@ function MainPageLayout.make(frame)
 				classes = {'navigation-cards'},
 				children = Array.map(WikiData.navigation, MainPageLayout._makeNavigationCard)
 			},
-			table.concat(MainPageLayout._makeCells(layout)),
+			MainPageLayout._makeCells(layout),
 		},
 	}
 end
 
+---@param body (string|Widget|Html|nil)|(string|Widget|Html|nil)[]
+---@return (string|Widget|Html|nil)|(string|Widget|Html|nil)[]
+function MainPageLayout._processCellBody(body)
+	local frame = mw.getCurrentFrame()
+	return type(body) == 'string' and frame:preprocess(body) or body
+end
+
 ---@param cells table[]
----@return string[]
+---@return Widget
 function MainPageLayout._makeCells(cells)
-	local frame = mw.getCurrentFrame()
 	local output = {}
 
-	table.insert(output, Grid._start_grid{})
 	for _, column in ipairs(cells) do
 		local cellContent = {}
 		for _, item in ipairs(column.children) do
 			local content = {}
 			if item.content then
+				local contentBody = item.content.body
 				if item.content.noPanel then
-					table.insert(content, frame:preprocess(item.content.body))
+					table.insert(content, MainPageLayout._processCellBody(contentBody))
 				else
-					table.insert(content, tostring(PanelWidget{
-						body = frame:preprocess(item.content.body),
+					table.insert(content, PanelWidget{
+						body = MainPageLayout._processCellBody(contentBody),
```

Comment:
Can we use the standard children here for input? (updating PanelWidget obviously too)

Reply:
possible implementation via #5546 (stacked on this PR so if @ElectricalBoy likes it he can just merge it into this PR)